wallet/lib/utils/favorites_change_notifier.dart

```diff
@@ -5,22 +5,20 @@ import '../model/nft.dart';
 import '../services/repository/repository.dart';
 import 'enums.dart';
 
-class FavoritesChangeNotifier extends ChangeNotifier{
-
+class FavoritesChangeNotifier extends ChangeNotifier {
   final Repository repository;
   FavoritesChangeNotifier({required this.repository});
 
   List<NFT> favorites = [];
 
-  Future<void> onInit() async {
-
-  }
+  Future<void> onInit() async {}
 
-  Future<void> addToFavorites({required FavoritesModel favoritesModel})async{
+  Future<void> addToFavorites({required FavoritesModel favoritesModel}) async {
     if (favoritesModel.type == NftType.TYPE_ITEM.name) {
       final item = await repository.getItem(cookBookId: favoritesModel.cookbookId, itemId: favoritesModel.id);
       if (item.isRight()) {
-        favorites.add(await NFT.fromItem(item.toOption().toNullable()!));
+        final nft = await NFT.fromItem(item.toOption().toNullable()!);
```

Comment:
Why create a variable when we can add an exclamation sign at the end... I would suggest do not create a variable just putting an exclamation sign at the end.

Reply:
If any error happens in this case `favorites.add(await NFT.fromItem(item.toOption().toNullable()!));` The stacktrace will arise from `NFT.fromItem` which is totally wrong. If we use this ` final nft = await NFT.fromItem(item.toOption().toNullable()!);` then the error will happen from the `onInit` Method which is where it happened.